### ISSUE TITLE
Discussion: Remove input_quantizer

### ISSUE THREAD

**Issue by lgeiger** [plumerai/larq#77]
After showing the library to a few researchers, it seams like the concept of an `input_quantizer` (essential a pre-activation) is something to get used to and slightly confusing at first.

Since an `input_quantizer` is essential an activation of the previous layer quantizers can already be used as normal activations after a layer `tf.keras.Activations("ste_sign")`.

I personally I strongly believe we should not encourage the use of two different concepts in our library (pre-activation quantization and (post-)activation quantization) since it adds unnecessary complexity. This would leave us with two options:

## Option 1: Keep the current API

This makes it a bit clearer which layer only uses binary operations. We should then stress the purpose of `input_quantizers` in the documentation since it is a non-standard concept.

### BNN without BatchNorm

```python
lq.layers.QuantDense(512, kernel_quantizer="ste_sign", kernel_constraint="weight_clip"),
lq.layers.QuantDense(10, input_quantizer="ste_sign", kernel_quantizer="ste_sign", kernel_constraint="weight_clip")
```

### BNN with BatchNorm

```python
lq.layers.QuantDense(512, kernel_quantizer="ste_sign", kernel_constraint="weight_clip"),
tf.keras.layers.BatchNorm(),
lq.layers.QuantDense(10, input_quantizer="ste_sign", kernel_quantizer="ste_sign", kernel_constraint="weight_clip")
```

## Option 2: Remove `input_quantizer` and use activations 

This is a simpler since it removes one new concept from the API and makes it clearer where the binarization happens without needing any docs (or implementation details) to explain.

### BNN without BatchNorm

```python
lq.layers.QuantDense(512, kernel_constraint="weight_clip", activation="ste_sign"),
lq.layers.QuantDense(10, kernel_quantizer="ste_sign", kernel_constraint="weight_clip")
```

### BNN with BatchNorm

```python
lq.layers.QuantDense(512, kernel_constraint="weight_clip"),
tf.keras.layers.BatchNorm(),
tf.keras.layers.Activation("ste_sign"),
lq.layers.QuantDense(10, kernel_quantizer="ste_sign", kernel_constraint="weight_clip")
```

Personally I am very much in favor of option two, because I think it is easier to understand and needs less `kwargs`, but I might miss something.

**Comment by lgeiger** [comment]
We opted for Option 1 due to the following reasons:
- Using pre-activation quantization makes it clear which layers and which operations are executed in a quantized way and reduces the amount of accidental bugs that can be introduced when adding layers that introduce additional floating point operations
- When constructing networks with skip connections like ResNets or DenseNets it can be usefull to have the option to quantize intermediate activations as well, therefor it makes sense to keep the possibility of using quantizers as normal activations.
